CanvasPlusPlayground/Common/Network/API Requests/GetAnnouncementsBatchRequest.swift

```
@@ -0,0 +1,93 @@
+//
+//  GetAnnouncementsBatchRequest.swift
+//  CanvasPlusPlayground
+//
+//  Created by Rahul on 1/19/25.
+//
+
+import Foundation
+
+struct GetAnnouncementsBatchRequest: CacheableArrayAPIRequest {
+    typealias Subject = AnnouncementAPI
+
+    var path: String { "announcements" }
+
+    var queryParameters: [QueryParameter] {
+        [
+            ("start_date", startDate?.ISO8601Format()),
+            ("end_date", endDate?.ISO8601Format()),
+            // ("active_only", activeOnly),  ONLY FOR TEACHERS
+            ("latest_only", latestOnly),
+            ("per_page", perPage)
+        ]
+        + contextCodes.map { ("context_codes[]", $0) }
+        + include.map { ("include[]", $0) }
+    }
+
+    // MARK: Query Params
+    /// At least one context code must be provided here otherwise request fails
+    let contextCodes: [String]
+    let startDate: Date?
+    let endDate: Date?
+    // let activeOnly: Bool? ONLY FOR TEACHERS
+    let latestOnly: Bool?
+    let include: [String]
+    let perPage: Int
+
+    init(
+        courseIds: [String],
+        startDate: Date? = nil,
+        endDate: Date? = nil,
+        latestOnly: Bool? = nil,
+        include: [String] = [],
+        perPage: Int = 50
+    ) {
+        self.contextCodes = courseIds.map { "course_\($0)" }
+        self.startDate = startDate
+        self.endDate = endDate
+        self.latestOnly = latestOnly
+        self.include = include
+        self.perPage = perPage
+    }
```

can you throw an error here if courseIds is empty? the api requires that courseIds be non-empty else it fails

---

I'll add an assert

---

perf